mllib/src/main/scala/org/apache/spark/ml/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -113,6 +113,10 @@ private[spark] object DecisionTreeMetadata extends Logging {
       throw new IllegalArgumentException(s"DecisionTree requires size of input RDD > 0, " +
         s"but was given by empty one.")
     }
+    if (numFeatures == 0) {
```

Comment:
would having two different error messages for the two cases be preferred?  It seems that we want to communicate to the user that either:
1.) the input RDD is empty, or
2.) the input is not empty but the features vector is empty
I will take the second suggestion and update with the require